feature/interface/tunnel/otg_tests/tunnel_based_gre_encap/README.md

```diff
@@ -0,0 +1,155 @@
+# TUN-1.1: Interface-based traffic IPv4 GRE Encapsulation using next-hop redirection
+
+## Summary
+
+The test verifies traffic encapsulation using a IPv4 tunnel GRE interface using next-hop redirection via policy forwarding.
+
+## Testbed type
+
+*  [`featureprofiles/topologies/atedut_4.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_4.testbed)
+
+## Procedure
+
+### Test environment setup
+
+*   DUT has an ingress port and 2 egress ports.
+
+    ```
+                             |         | ---- | ATE Port 2 |
+        [ ATE Port 1 ] ----  |   DUT   |      |            |
+                             |         | ---- | ATE Port 3 |
+    ```
+
+*   Traffic is generated from ATE Port 1.
+*   ATE Port 2 is used as the destination port for encapsulated 
+    traffic.
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+
+#### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces. Configure MTU of 9216 (L2) on ATE Port1, Port2
+ 
+2.  IPv4 and IPv6 static routes to test destination networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  A set of IPv4 static routes to 32x IPv4 GRE tunnel endpoints towards ATE Port 2.
+
+4.  32 tunnel interfaces are configured wtih IPv4 tunnel endpoints as destination and device loopback as source. Configure MTU on tunnel interfaces to 2000.
+    All tunnel interfaces should be unnumbered referencing loopback interface.
+
+5.  IPv4 encapsulation next-hop (IPV4-ENCAP-NH) is allocated. A set of IPv4 static routes to IPV4-ENCAP-NH is configured toward 32 tunnel interfaces.
+
+6.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and set next-hop to IPV4-ENCAP-NH.
+    - Match IPV6-SRC2 and set next-hop to IPV4-ENCAP-NH.
+
+7.  Configure QoS classifier for incoming traffic on ATE Port1 for IPv4 and IPv6 traffic. 
+     QoS classifier should remark egress packet to the matching ingress DSCP value (eg. match DSCP 32, set egress DSCP 32).
+     Match and remark all values for 3 leftmost DSCP bits [0, 8, 16, 24, 32, 40, 48, 56].
+    
+
+### TUN-1.1.1: Verify PF GRE encapsulate action for IPv4 traffic
+Generate traffic on ATE Port 1 from IPV4-SRC2 from a random combination of 1000 source addresses to IPV4-DST at linerate.
+Use 512 bytes frame size.
```

Comment:
Is the topology specified with only one ingress port suitable for the performance requirement?  I would expect a  NPU specific or NPU relevant number of ports needs to be specified, specific port speeds and a duration.    (by default, if you run an ondatra test with 3 ports, the binding/reservation system will chose whatever is available). Do you care if this is 10G, 100G, 400G, etc?

(Nit:  For functionality, only a few packets should really be required.  If we really need a performance tests, it could be  separate  subtests.  It would be ok to say, repeat subtests 1,2,3 with 1B packets at line rate since the code would likely be written that way). The goal being to very quickly see if there is a functionality problem vs. a performance problem by (sub)test id. )